CodeMaid/Helpers/FileHeaderHelper.cs

```diff
@@ -0,0 +1,143 @@
+                        GetHeaderLength(docStart, "/*", "*/");
+
+                case CodeLanguage.HTML:
+                case CodeLanguage.XAML:
+                case CodeLanguage.XML:
+                    return GetHeaderLength(docStart, "<!--", "-->");
+
+                case CodeLanguage.CSS:
+                    return GetHeaderLength(docStart, "/*", "*/");
+
+                case CodeLanguage.PHP:
+                    return GetHeaderLength(docStart, "//") +
+                        GetHeaderLength(docStart, "#") +
+                        GetHeaderLength(docStart, "/*", "*/");
+
+                case CodeLanguage.PowerShell:
+                    return GetHeaderLength(docStart, "#") +
+                        GetHeaderLength(docStart, "<#", "#>");
+
+                case CodeLanguage.R:
+                    return GetHeaderLength(docStart, "#");
+
+                case CodeLanguage.FSharp:
+                    return GetHeaderLength(docStart, "//") +
+                        GetHeaderLength(docStart, "(*", "*)");
+
+                case CodeLanguage.VisualBasic:
+                    return GetHeaderLength(docStart, "'");
+
+                case CodeLanguage.JSON:
+                case CodeLanguage.Unknown:
+                default:
+                    return 0;
+            }
+        }
+
+        internal static int GetHeaderLength(string docStart, string commentSyntax)
+        {
+            if (!docStart.TrimStart().StartsWith(commentSyntax))
+            {
+                return 0;
+            }
+
+            var separator = new string[] { Environment.NewLine };
+            var lines = docStart.Split(separator, StringSplitOptions.None);
+            var header = new List<string>();
+
+            // adds starting empty lines
+            foreach (var line in lines)
+            {
+                if (!string.IsNullOrWhiteSpace(line))
+                {
+                    break;
+                }
+
+                header.Add(line);
+            }
+
+            // adds comment lines
+            foreach (var line in lines.ToList().Skip(header.Count))
+            {
+                if (!line.StartsWith(commentSyntax))
+                {
+                    break;
+                }
+
+                header.Add(line);
+            }
+
+            var nbChar = 0;
+            header.ToList().ForEach(x => nbChar += x.Length + 1);
+
+            return nbChar;
+        }
+
+        internal static int GetHeaderLength(string docStart, string commentSyntaxStart, string commentSyntaxEnd)
+        {
+            if (!docStart.TrimStart().StartsWith(commentSyntaxStart))
+            {
+                return 0;
+            }
+
+            var startIndex = docStart.IndexOf(commentSyntaxStart);
+            var endIndex = docStart.IndexOf(commentSyntaxEnd);
+            var emptyStart = docStart.Substring(0, startIndex);
+            var emptyStartLength = emptyStart.Length - Regex.Matches(emptyStart, Environment.NewLine).Count;
```

Comment:
I'm not quite sure what this logic is handling.. it looks like its reducing the character length based on leading newlines?

Reply:
You're right, my logic was flawed. I fixed and simplified it, fixed related unit tests and added comments and remarks trying to explain what's going on (counting only 1 character per end of line)